src/main/java/org/openstreetmap/atlas/checks/distributed/ShardedIntegrityChecksSparkJob.java

```diff
@@ -0,0 +1,434 @@
+            }
+
+            flagContainer.reconstructEvents().parallel().forEach(eventService::post);
+            eventService.complete();
+        };
+    }
+
+    /**
+     * {@link PairFunction} to run each {@link ShardedCheckFlagsTask} through to produce
+     * {@link org.openstreetmap.atlas.checks.flag.CheckFlag}s.
+     *
+     * @param input
+     *            {@link String} input folder path
+     * @param output
+     *            {@link String} output folder path
+     * @param configurationMap
+     *            {@link org.openstreetmap.atlas.generator.tools.spark.SparkJob} configuration map
+     * @param fileHelper
+     *            {@link SparkFileHelper}
+     * @param sharding
+     *            spark {@link Broadcast} of the current {@link Sharding}
+     * @param shardDistanceExpansion
+     *            {@link Distance} to expand the shard group
+     * @param multiAtlas
+     *            boolean whether to use a multi or dynamic Atlas
+     * @return {@link PairFunction} that takes {@link ShardedCheckFlagsTask} and returns a
+     *         {@link Tuple2} of a {@link String} country code and {@link UniqueCheckFlagContainer}
+     */
+    @SuppressWarnings("unchecked")
+    private PairFunction<ShardedCheckFlagsTask, String, UniqueCheckFlagContainer> produceFlags(
+            final String input, final String output, final Map<String, String> configurationMap,
+            final SparkFileHelper fileHelper, final Broadcast<Sharding> sharding,
+            final Distance shardDistanceExpansion, final boolean multiAtlas)
+    {
+        return task ->
+        {
+            // Get the atlas
+            final Function<Shard, Optional<Atlas>> fetcher = this.atlasFetcher(input,
+                    task.getCountry(), configurationMap);
+            final Atlas atlas;
+
+            // Use dynamic or multi atlas (multi runs faster locally)
+            if (multiAtlas)
+            {
+                atlas = new MultiAtlas(
+                        StreamSupport
+                                .stream(sharding.getValue()
+                                        .shards(task.getShard().bounds()
+                                                .expand(shardDistanceExpansion))
+                                        .spliterator(), true)
+                                .map(fetcher).filter(Optional::isPresent).map(Optional::get)
+                                .collect(Collectors.toList()));
+            }
+            else
+            {
+                final DynamicAtlasPolicy policy = new DynamicAtlasPolicy(fetcher,
+                        sharding.getValue(), Collections.singleton(task.getShard()),
+                        task.getShard().bounds().expand(shardDistanceExpansion))
+                                .withDeferredLoading(true).withAggressivelyExploreRelations(true)
+                                .withExtendIndefinitely(false);
```

Comment:
Wondering how useful it'd be to have a configurable DynamicAtlasPolicy, or one defined in a file

Reply:
This is definitely something I want to have eventually. That may be part of the solution for dealing with large relations. For now I think it is fine to have it hard coded while that solution is architected.